source/system/structs/VectorList.ooc

```diff
@@ -51,9 +51,16 @@ VectorList: class <T> extends List<T>{
 		this _count -= 1
 		result
 	}
-	removeAt: override func (index: Int) {
-		this _vector copy(index + 1, index)
-		this _count -= 1
+	removeAt: override func (index: Int) { this removeAt(index, index + 1) }
+	removeAt: func ~range (range: Range) { this removeAt(range min, range max) }
+	removeAt: func ~indices (start, end: Int) {
+		count := end - start
+		version(safe) {
+			raise(count < 0 || start > this count || end > this count,
+				"Invalid range in VectorList::removeAt: start=%d end=%d (list count=%d)" format(start, end, this count))
```

Comment:
This will leak memory, though, because in `-Dsafe` the string will be created regardless of whether the argument is true or not.

Better perhaps to use `if (count < 0 || ...) raise("Invalid range..." format(...))` which will only create the string if we're raising, at which point leaks don't matter.